src/m365/pa/commands/environment/environment-get.ts

```diff
@@ -0,0 +1,61 @@
+import { Logger } from '../../../../cli';
+import {
+    CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import AzmgmtCommand from '../../../base/AzmgmtCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  name: string;
+}
+
+class PaEnvironmentGetCommand extends AzmgmtCommand {
+  public get name(): string {
+    return commands.ENVIRONMENT_GET;
+  }
+
+  public get description(): string {
+    return 'Gets information about the specified Microsoft Power Apps environment';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving information about Microsoft Power Apps environment ${args.options.name}...`);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}providers/Microsoft.PowerApps/environments/${encodeURIComponent(args.options.name)}?api-version=2016-11-01`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    request
+      .get(requestOptions)
+      .then((res: any): void => {
+        logger.log(res);
+
+        cb();
+      }, (rawRes: any): void => this.handleRejectedODataJsonPromise(rawRes, logger, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-n, --name <name>'
+      }
+    ];
```

Comment:
The name is required but the script is not giving me a warning if I do not specify a value.

I am receiving a 404 from the service

Reply:
Which command did you run exactly?

```sh
m365 pa environment get
```

or

```sh
m365 pa environment get --name
```

If the latter, then `name` is defined and its value is set to `true`

Reply:
Cant reproduce this issue on Windows or Linux other then with how @waldekmastykarz described it.